### PR TITLE
glassfish/issues/24059 exclude ManagedExecutorDefinitionWebTests#testCopyCompletableFutureEJB test from  run and only include that test in the run

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>

**Fixes Issue**
The idea is to fix https://github.com/eclipse-ee4j/glassfish/issues/24059 by excluding the `ManagedExecutorDefinitionWebTests#testCopyCompletableFutureEJB` test from the `suite-web.xml` run but do include the `ManagedExecutorDefinitionWebTests#testCopyCompletableFutureEJB` test in the (new) `suite-web-group2.xml`.

The `suite-web-group2.xml` should only run the one test that previously failed when run with all of the Web Profile tests (see the issue comments for more details).

### CODE REVIEW

**Comment by scottmarlow** [comment]
Will test via https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run and adjust as needed.

**Comment by gurunrao** [comment]
PR looks good to me, will wait for final version.

File: glassfish-runner/concurrency-tck/suite-web-group2.xml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                </methods>
+            </class>
+            <class name="ee.jakarta.tck.concurrent.spec.signature">
+                <methods>
+                    <exclude name=".*"></exclude>
+                </methods>
+            </class>
+            
+        </classes>
+    </test>
```
Comment (OndroMih) [inline glassfish-runner/concurrency-tck/suite-web-group2.xml:210]: It should be enough to just include the single test, or not? Such as:
```suggestion
    <test name="jakarta-concurrency.tck">
        
        <classes>
            <class name="ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef.ManagedExecutorDefinitionWebTests">
                <methods>
                    <include name="testCopyCompletableFutureEJB"></include>
                </methods>
            </class>
            
        </classes>
    </test>
```

**Review by OndroMih — APPROVED** [review]
As I suggested, I think the TestNG config for the second suite can be much simpler. Otherwise this looks good.

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/5/ ran `web` which should of excluded `ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef.ManagedExecutorDefinitionWebTests.testCopyCompletableFutureEJB` but didn't.

**Comment by OndroMih** [comment]
The groups and packages section should be removed from `suite-web-group2.xml`. So that it looks like this:

```
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
<suite name="jakarta-concurrency" verbose="2" configfailurepolicy="continue">
     <listeners>
        <!-- Required - avoid randomly mixed test method execution -->
        <listener class-name="org.jboss.cdi.tck.impl.testng.SingleTestClassMethodInterceptor" />
        <!-- Optional - intended for debug purpose only -->
        <listener class-name="org.jboss.cdi.tck.impl.testng.ProgressLoggingTestListener"/>
        <!-- Optional - it's recommended to disable the default JUnit XML reporter -->
        <listener class-name="org.testng.reporters.SuiteHTMLReporter"/>
        <listener class-name="org.testng.reporters.FailedReporter"/>
        <listener class-name="org.testng.reporters.XMLReporter"/>
        <listener class-name="org.testng.reporters.EmailableReporter"/>
        <listener class-name="org.testng.reporters.TestHTMLReporter" />
    </listeners>
    <test name="jakarta-concurrency.tck">
        <classes>
            <class name="ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef.ManagedExecutorDefinitionWebTests">
                <methods>
                    <include name="testCopyCompletableFutureEJB"></include>
                </methods>
            </class>
        </classes>
    </test>
</suite>
```

There's no need to specify groups or packages if we just want to execute a single test method.

**Comment by scottmarlow** [comment]
> The groups and packages section should be removed from `suite-web-group2.xml`. So that it looks like this:
> 
> ```
> <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
> <suite name="jakarta-concurrency" verbose="2" configfailurepolicy="continue">
>      <listeners>
>         
>         <listener class-name="org.jboss.cdi.tck.impl.testng.SingleTestClassMethodInterceptor" />
>         
>         <listener class-name="org.jboss.cdi.tck.impl.testng.ProgressLoggingTestListener"/>
>         
>         <listener class-name="org.testng.reporters.SuiteHTMLReporter"/>
>         <listener class-name="org.testng.reporters.FailedReporter"/>
>         <listener class-name="org.testng.reporters.XMLReporter"/>
>         <listener class-name="org.testng.reporters.EmailableReporter"/>
>         <listener class-name="org.testng.reporters.TestHTMLReporter" />
>     </listeners>
>     <test name="jakarta-concurrency.tck">
>         <classes>
>             <class name="ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef.ManagedExecutorDefinitionWebTests">
>                 <methods>
>                     <include name="testCopyCompletableFutureEJB"></include>
>                 </methods>
>             </class>
>         </classes>
>     </test>
> </suite>
> ```
> 
> There's no need to specify groups or packages if we just want to execute a single test method.

Thanks @OndroMih , trying this now.  [It worked, wonderful](https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/org.glassfish$glassfish.concurrency-tck/8/testReport/)! :-)

Any suggestions for improving my change to the `suite-web.xml` file?  From https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/5/ results, the failing test wasn't ignored by change:
```
            <class name="ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef.ManagedExecutorDefinitionWebTests">
                <methods>
                    <exclude name="testCopyCompletableFutureEJB"></exclude>
                </methods>
            </class>
```

**Comment by OndroMih** [comment]
I executed the TCk locally and the test is excluded. I think that the Jenkins job overwrites the TestNG suite XML files with those in the TCK. I see the following lines in the output of the Jenkins job:

```

+ cp suite-web.xml /home/jenkins/agent/workspace/10/hackjakarta-concurrency-tck-glassfish-run/glassfish-runner/concurrency-tck/
+ cp suite.xml /home/jenkins/agent/workspace/10/hackjakarta-concurrency-tck-glassfish-run/glassfish-runner/concurrency-tck/
```

**Comment by OndroMih** [comment]
I've edited your Jenkins job (removed the instructions to copy the suite.xml files) and restarted the job here: https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/9/.

**Comment by scottmarlow** [comment]
> I've edited your Jenkins job (removed the instructions to copy the suite.xml files) and restarted the job here: https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/9/.
> 
> The job finished with 147 tests passed, no failures: https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/org.glassfish$glassfish.concurrency-tck/9/testReport/.
> 
> So we're can merge this PR and get ready to submit a CCR for EE 10 Web Profile with GF 7.0-M7 I guess.

Thanks, just merged it.

**Comment by scottmarlow** [comment]
Hmm, the https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run/ job has:

```
cp suite-web.xml $WORKSPACE/glassfish-runner/concurrency-tck/
cp suite.xml $WORKSPACE/glassfish-runner/concurrency-tck/
```

I'm not sure that removing the file copies is right but am running https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/10 with just the `suite-web-group2.xml` option to see what happens.

**Comment by scottmarlow** [comment]
> I'm not sure that removing the file copies is right but am running https://ci.eclipse.org/jakartaee-tck/job/10/job/hackjakarta-concurrency-tck-glassfish-run/10 with just the `suite-web-group2.xml` option to see what happens.

The `concurrency-tck-glassfish-run/10` test passed (only the single test was run).  I'll update https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run/ + the jdk17 variation to have the same changes.

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run/104 - is running with PROFILE=`webgroup2`

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run/105 is running with PROFILE=`web`

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run/106/ passed also with PROFILE=`full`.

**Comment by scottmarlow** [comment]
https://ci.eclipse.org/jakartaee-tck/job/10/job/jakarta-concurrency-tck-glassfish-run-jdk17/ also passed the 3 PROFILE variations.  We had one transient failure but good otherwise.  I'm going to mark the issue as solved.

**Pull request merged**